lib/BuildSystem/BuildSystem.cpp

```diff
@@ -208,6 +208,9 @@ class BuildSystemImpl : public BuildSystemCommandInterface {
 
   /// Flag indicating if the build has been cancelled.
   std::atomic<bool> isCancelled_{ false };
+
+  /// Cache of instantiated shell command handlers.
+  llvm::StringMap<std::unique_ptr<ShellCommandHandler>> shellHandlers;
```

Comment:
Shouldn't we be clearing this at the start of build()?   As it currently stands we will reuse these across multiple builds if the engine is cached.

Reply:
Oops, yes, indeed.